### PR TITLE
Technical/finish up package

### PR DESCRIPTION
**Pull request by pawelirh**
- compose
- readme
- workflow

### CODE REVIEW

File: .github/workflows/build-docker-image.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+name: Build/Publish Docker Image
+
+on:
+  push:
+    branches:
+      - "main"
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: "A tag that the image will be built with"
```
Comment (TopRamen1) [inline .github/workflows/build-docker-image.yaml]: Redundant tag declaration, should be changed to melodic or be deleted

File: demo/compose.ouster.yaml [hunk 2]
```diff
@@ -6,7 +6,7 @@ services:
     network_mode: host
     devices:
       - /dev/ptp0:/dev/ptp0
-    command: sudo ptp4l -i eno1 -m [global] tx_timestamp_timeout 10 
+    command: sudo ptp4l -i eno1 -m [global] tx_timestamp_timeout 10
```
Comment (TopRamen1) [inline demo/compose.ouster.yaml]: PTP synchronization form container did not work when testing and had to be replaced by a systemctl service. We should include that here.

File: demo/compose.rviz.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+services:
+
+  rviz:
+    image: husarion/rviz:noetic
+    restart: on-failure
+    network_mode: host
+    ipc: host
+    volumes:
+      - /tmp/.X11-unix:/tmp/.X11-unix:rw
+      - ./config/default.rviz:/root/.rviz2/default.rviz
```
Comment (TopRamen1) [inline demo/compose.rviz.yaml]: The empty default file should be replaced with config containing at least ouster pointcloud by default.

**Review by TopRamen1 — CHANGES_REQUESTED** [review]

File: .github/workflows/build-docker-image.yaml (hunk 1 above)
Comment (pawelirh) [reply .github/workflows/build-docker-image.yaml]: I just removed workflow_dispatch

File: demo/compose.ouster.yaml (hunk 2 above)
Comment (pawelirh) [reply demo/compose.ouster.yaml]: ✅

File: demo/compose.rviz.yaml (hunk 3 above)
Comment (pawelirh) [reply demo/compose.rviz.yaml]: Since we don't have access to the ouster, the follow-up task "add the default rviz config" will be created

Comment (TopRamen1) [reply demo/compose.rviz.yaml]: :+1:

**Review by TopRamen1 — APPROVED** [review]

**Comment by pkowalsk1** [comment]
Due to the reorganization of the ouster-lidar repositories (https://github.com/ouster-lidar), the Dockerfile and demo folder need the additional changes proposed in the next commit

File: demo/compose.ouster.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+services:
+
+  ouster-lidar:
+    image: husarion/ouster:melodic
+    restart: unless-stopped
+    network_mode: host
```
Comment (KmakD) [inline demo/compose.ouster.yaml]: missing `container_name`

File: Dockerfile [hunk 5]
```diff
@@ -4,34 +4,27 @@ FROM ros:${ROS_DISTRO}-ros-core
...
+    python-rospkg 
+
+WORKDIR /ros_ws
+
+RUN git clone --recurse-submodules https://github.com/ouster-lidar/ouster-ros.git src/ouster-ros
+
+# RUN apt-get update && \
+RUN rosdep init && \
+    rosdep update --rosdistro=$ROS_DISTRO && \
+    rosdep install -y --from-paths src && \
```
Comment (KmakD) [inline Dockerfile]:
```suggestion
    rosdep install --from-paths src --ignore-src -y && \
```
lets ignore packages from source

File: Dockerfile [hunk 6]
```diff
@@ -4,34 +4,27 @@ FROM ros:${ROS_DISTRO}-ros-core
...
+    source /opt/ros/$ROS_DISTRO/setup.bash && \
+    catkin_make -DCATKIN_ENABLE_TESTING=0 -DCMAKE_BUILD_TYPE=Release
```
Comment (KmakD) [inline Dockerfile:30]: We can reduce step size by adding
```
apt clean && \
rm -rf /var/lib/apt/lists/*
```

File: Dockerfile [hunk 7]
```diff
@@ -4,34 +4,27 @@ FROM ros:${ROS_DISTRO}-ros-core
...
+RUN apt-get update && apt-get install -y \
+    build-essential \
+    cmake \
+    curl \
+    debhelper \
+    dpkg-dev \
+    fakeroot \
+    python-bloom \
+    python-rosdep \
+    python-rospkg
```
Comment (KmakD) [inline Dockerfile]:
```suggestion
RUN apt-get update && apt-get install -y \
        build-essential \
        cmake \
        curl \
        debhelper \
        dpkg-dev \
        fakeroot \
        python-bloom \
        python-rosdep \
        python-rospkg
```
I prefer it more indent as it is still part of the above command

**Review by KmakD — CHANGES_REQUESTED** [review]
Some small changes

**Review by KmakD — APPROVED** [review]

**Pull request merged**